cvat-ui/src/components/annotation-page/review/use-dialog-positioning.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { useEffect, useState, RefObject } from 'react';
+
+interface UseDialogPositioningParams {
+    ref: RefObject<HTMLDivElement>;
+    top: number;
+    left: number;
+    scale: number;
+    angle: number;
+    clientCoordinates: [number, number];
+    canvasRect: DOMRect | null;
+}
+
+interface UseDialogPositioningResult {
+    adjustedTop: number;
+    adjustedLeft: number;
+}
+
+export function useDialogPositioning({
```

Comment:
IssueAggregator creates new `clientCoordinates` each render and passes it to children.
Children (CreateIssueDialog/IssueDialog) accepts `clientCoordinates` and passes it to this hook. 
Each time it causes effect as `clientCoordinates` is dependency. 

So, the question is. Should it really be a hook or just a function is enough? From my perspective, it may be a simple function without useState/useEffect. That would be simpler and in current implementation we do not have any benefits of using hook

Reply:
It could just be a function, but the main issue is that before adjusting `top` and `left`, we need to know the dimensions of the issue dialog. That means we still need a hook that runs once `dialogRef.current` is not `null`, and then calls the coordinate transformation function.

As a result, this hook would end up being duplicated in both `CreateIssueDialog` and `IssueDialog`.